### ISSUE TITLE
🐞 Threshold-based `predict_proba` returns incorrect mask for nonzero ambiguity threshold

### ISSUE THREAD

**Issue by sanjaradylov** [sanjaradylov/scikit-fallback#2]
**Describe the bug**
`ThresholdFallbackClassifier` and `ThresholdFallbackClassifierCV` return an `FBNDArray` w/ a mask produced by only a global threshold ignoring ambiguity threshold.

**To Reproduce**
```python
>>> import numpy  as np
>>> from skfb.estimators import ThresholdFallbackClassifier
>>> from sklearn.linear_model import LogisticRegression
>>> X = np.array([[0, 0], [4, 4], [1, 1], [3, 3], [2.5, 2], [2.0, 2.5]])
>>> y = np.array([0, 1, 0, 1, 0, 1])
>>> rejector = ThresholdFallbackClassifier(LogisticRegression(random_state=0), ambiguity_threshold=0.2)
>>> rejector.fit(X, y).predict_proba(X).get_dense_fallback_mask()
array([False, False, False, False, False, False])
```

**Expected behavior**
The last two examples should be rejected because of close class probabilities given high ambiguity threshold.

**Screenshots**
NA

**Please complete the following information:**
 - OS: NA
 - Python version: NA
- `scikit-fallback` version: 0.0.0

**Additional context**
NA
